libs/dfi/README.md

```diff
@@ -19,4 +19,243 @@ See the License for the specific language governing permissions and
+
+~~~
+:header
+type=interface
+name=calculator
+version=1.0.0
+:annotations
+classname=org.example.Calculator
+:types
+StatsResult={DDD[D average min max input}
+:methods
+add(DD)D=add(#am=handle;PDD#am=pre;*D)N
+~~~
+
+As above, the interface description has four sections: the header section, annotations section, types section, and methods section. The format of each section is as follows:
+
+~~~
+ ':header\n' 'Name=Value\n'...
+ ':annotations\n' 'Name=Value\n'...
+ ':types\n' 'TypeId=Value\n'...
+ ':methods\n' 'MethodId=Value\n'...
+~~~
+
+Among them, the legal characters that can be used in the “name” and “TypeId” include [a-zA-Z0-9_], the legal characters in “MethodId” include [a-zA-Z0-9_] and ".();[{}/". Besides [a-zA-Z0-9], the legal characters in “value” also include ".<>{}[]?;:~!@#$%^&*()_+-=,./'". It's worth noting that there should not be spaces on either side of '=', and each statement must end with a newline(‘\n’).
+
+For the interface description, its header section must include three elements: type, name, version. The value of "type" should be "interface", "name" is the interface name (service name), the value of "version" should coincide with the version number in the actually used interface header file, and it should conform to [semantic versioning requirements](https://semver.org/).
+
+#### The Data Types In The Interface Descriptor
+
+The data types supported by the interface description include:
+
+- **Simple Types**
+
+  *Type schema*:
+  
+  |**Identifier**|B  |D     |F    |I      |J      |S      |V   |Z             |b    | i      | j      | s      |P     |t     |N  | 
+  |---------|---|------|-----|-------|-------|-------|----|--------------|-----|--------|--------|--------|------|------|---|
+  |**Types**|char|double|float|int32_t|int64_t|int16_t|void|boolean(uint8)|uchar|uint32_t|uint64_t|uint16_t|void *|char *|int|
+
+
+- **Complex Types(Struct)**
+
+  *Type schema*:
+  ~~~
+  {[Type]+ [(Name)(SPACE)]+}
+  ~~~
+  *Example*:
+  ~~~
+  {DDII a b c d}
+  ~~~
+  *To C language*:
+  ~~~
+  struct { double a; double b; int c; int d; };
+  ~~~
+
+- **Sequence Type**
+
+  *Type schema*:
+  ~~~
+  [(type)
+  ~~~
+  *Example*:
+  ~~~
+  [D
+  ~~~
+  *To C language*:
+  ~~~
+  struct {
+    uint32_t cap;
+    uint32_t len;
+    duoble *buf;
+  };
+  ~~~
+  
+- **Typed Pointer**
+
+  *Type schema*:
+  ~~~
+  *(Type)
+  ~~~
+  *Example*:
+  ~~~
+  *D
+  ~~~
+  *To C language*:
+  ~~~
+  duoble *d;
+  ~~~
+
+- **Reference By Value**
+
+  *Type schema*:
+  ~~~
+  l(name);
+  ~~~
+  *Example*:
+  ~~~
+  MySubType={jDD time d1 d2}
+  MyType={DDlMySubType; d11 d12 subTypeVal}
+  ~~~
+  *To C language*:
+  ~~~
+  struct MySubType{
+    uint64_t time;
+    double d1;
+    double d2;
+  };
+  struct MyType {
+    double d11;
+    double d12;
+    struct MySubType subTypeVal;
+  };
+  ~~~
+
+- **Pointer Reference**
+
+  *Type schema*:
+  ~~~
+  L(name);//shortcut for *l(name);
```

Comment:
Now that we have `*`, why we need `*l`?

Reply:
Given the support for `*`, I think we can drop `L` and `t` so that we do not have alternatives so "say" the same thing.